hyperframe/frame.py

```diff
@@ -110,7 +113,15 @@ def parse_frame_header(header):
         stream_id = fields[4]
 
         if type not in FRAMES:
-            raise UnknownFrameError(type, length)
+            if strict:
+                raise UnknownFrameError(type, length)
+
+            frame = ExtensionFrame(
+                type=type,
+                flag_byte=flags,
+                stream_id=stream_id,
+            )
+            return (frame, length)
```

Comment:
So, for the sake of clarity, let's actually do this:

```python
try:
    frame = FRAMES[type](stream_id)
except KeyError:
    if strict:
        raise UnknownFrameError(type, length)
    frame = ExtensionFrame(type=type, stream_id=stream_id)
```

Then, the flag byte can end up in place using `parse_flags`, and we can reduce the amount of return points.